src/main/scala/ru/gaptrakhmanov/telegram/helper/commands/wordgame/state/WordGameState.scala

```diff
@@ -0,0 +1,50 @@
+package ru.gaptrakhmanov.telegram.helper.commands.wordgame.state
+
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import ru.gaptrakhmanov.telegram.helper.commands.wordgame.dto.{GameInfo, User, UsersAnswers, WordSet}
+
+trait WordGameState[F[_]] {
+
+  def saveWordSet(user: User, words: WordSet): F[Unit]
+
+  def getWordSet(user: User): F[WordSet]
+
+  def isEmptySet(user: User): F[Boolean]
+
+  def resetWordSet(user: User): F[Unit]
+
+  def saveUserAnswer(user: User, word: String): F[Unit]
+
+  def getUserAnswers(user: User): F[UsersAnswers]
+}
+
+object WordGameState {
+  def create[F[_] : Sync](): F[WordGameState[F]] =
+    Ref.of[F, Map[User, GameInfo]](Map.empty).map { ref =>
```

Comment:
It looks good to me, only advice is that sometimes we want to do multiple logic inside of a modify to get a sense of all the operations done in atomic way.